app/controllers/lessons/previews_controller.rb

```diff
@@ -1,8 +1,20 @@
 module Lessons
   class PreviewsController < ApplicationController
-    def show; end
+    def show
+      @preview = LessonPreview.find_or_initialize_by(id: params[:uuid])
+    end
 
     def create
+      preview_link = LessonPreview.create(lesson_preview_params)
+
+      if preview_link
+        render json: { preview_link: preview_link(preview_link.id) }
```

Comment:
Can we inline the link please?
```suggestion
        render json: { preview_link: lessons_preview_url(uuid: preview_link.id)  }
```